diff --git a/node_modules/image-size/dist/index.cjs b/node_modules/image-size/dist/index.cjs
index bde0210..a450e76 100644
--- a/node_modules/image-size/dist/index.cjs
+++ b/node_modules/image-size/dist/index.cjs
@@ -173,6 +173,7 @@ var HEIF = {
         width = rawWidth - cropRight;
       }
       images.push({ height, width });
+      if (ispeBox.size <= 0) break;
       currentOffset = ispeBox.offset + ispeBox.size;
     }
     if (images.length === 0) {
@@ -255,6 +256,7 @@ var ICNS = {
       const imageHeader = readImageHeader(input, imageOffset);
       const imageSize2 = getImageSize2(imageHeader[0]);
       images.push(imageSize2);
+      if (imageHeader[1] <= 0) break;
       imageOffset += imageHeader[1];
     }
     if (images.length === 0) {
@@ -490,6 +492,7 @@ function extractPartialStreams(input) {
     partialStreams.push(
       input.slice(jxlpBox.offset + 12, jxlpBox.offset + jxlpBox.size)
     );
+    if (jxlpBox.size <= 0) break;
     offset = jxlpBox.offset + jxlpBox.size;
   }
   return partialStreams;
diff --git a/node_modules/image-size/dist/index.mjs b/node_modules/image-size/dist/index.mjs
index 22c17af..4ad8f32 100644
--- a/node_modules/image-size/dist/index.mjs
+++ b/node_modules/image-size/dist/index.mjs
@@ -169,6 +169,7 @@ var HEIF = {
         width = rawWidth - cropRight;
       }
       images.push({ height, width });
+      if (ispeBox.size <= 0) break;
       currentOffset = ispeBox.offset + ispeBox.size;
     }
     if (images.length === 0) {
@@ -251,6 +252,7 @@ var ICNS = {
       const imageHeader = readImageHeader(input, imageOffset);
       const imageSize2 = getImageSize2(imageHeader[0]);
       images.push(imageSize2);
+      if (imageHeader[1] <= 0) break;
       imageOffset += imageHeader[1];
     }
     if (images.length === 0) {
@@ -486,6 +488,7 @@ function extractPartialStreams(input) {
     partialStreams.push(
       input.slice(jxlpBox.offset + 12, jxlpBox.offset + jxlpBox.size)
     );
+    if (jxlpBox.size <= 0) break;
     offset = jxlpBox.offset + jxlpBox.size;
   }
   return partialStreams;
